components/event-publisher-proxy/pkg/handler/nats/handler_test.go

```diff
@@ -0,0 +1,453 @@
+		opts := &options.Options{MaxRequestSize: 65536}
+		natsHandler := NewNatsHandler(messageReceiver, msgSender, test.natsConfig.RequestTimeout, legacyTransformer, opts, subscribedProcessor, test.logger)
+		assert.NotNil(t, natsHandler)
+		go func() {
+			if err := natsHandler.Start(ctx); err != nil {
+				t.Errorf("Failed to start handler with error: %v", err)
+			}
+		}()
+
+		// wait that the embedded servers are started
+		testingutils.WaitForHandlerToStart(t, healthEndpoint)
+
+		// prepare event type from subscription
+		assert.NotNil(t, subscription.Spec.Filter)
+		assert.NotEmpty(t, subscription.Spec.Filter.Filters)
+		eventType := subscription.Spec.Filter.Filters[0].EventType.Value
+
+		// publish a message to NATS and validate it
+		connection := connectToNatsOrFail(t, natsUrl)
+		validator := validateNatsSubject(t, expectedNatsSubject)
+		subscribeToEventOrFail(t, connection, eventType, validator)
+
+		// run the tests for publishing cloudevents
+		for _, testCase := range handlertest.TestCasesForCloudEvents {
+			t.Run(testCase.Name, func(t *testing.T) {
+				body, headers := testCase.ProvideMessage()
+				resp, err := testingutils.SendEvent(publishEndpoint, body, headers)
+				if err != nil {
+					t.Errorf("Failed to send event with error: %v", err)
+				}
+				_ = resp.Body.Close()
+				if testCase.WantStatusCode != resp.StatusCode {
+					t.Errorf("Test failed, want status code:%d but got:%d", testCase.WantStatusCode, resp.StatusCode)
+				}
+			})
+		}
+	}
+
+	// make sure not to change the cloudevent, even if its event-type contains none-alphanumeric characters
+	exec(t, testingutils.ApplicationName, testingutils.CloudEventTypeNotClean)
+	exec(t, testingutils.ApplicationNameNotClean, testingutils.CloudEventTypeNotClean)
+}
+
+func TestNatsHandlerForLegacyEvents(t *testing.T) {
+	exec := func(t *testing.T, applicationName string, expectedNatsSubject string) {
+		// test environment
+		var (
+			healthEndpoint        = fmt.Sprintf("http://localhost:%d/healthz", test.natsConfig.Port)
+			publishLegacyEndpoint = fmt.Sprintf("http://localhost:%d/%s/v1/events", test.natsConfig.Port, applicationName)
+		)
+
+		test.logger.Info("TestNatsHandler started")
+
+		// a cancelable context to be used
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		// configure message receiver
+		messageReceiver := receiver.NewHttpMessageReceiver(test.natsConfig.Port)
+		assert.NotNil(t, messageReceiver)
+
+		// create a Nats sender
+		natsUrl := test.natsServer.ClientURL()
+		assert.NotEmpty(t, natsUrl)
+		msgSender := sender.NewNatsMessageSender(ctx, natsUrl, test.logger)
+
+		// configure legacyTransformer
+		appLister := handlertest.NewApplicationListerOrDie(ctx, applicationName)
+		legacyTransformer := legacy.NewTransformer(
+			test.natsConfig.ToConfig().BEBNamespace,
+			test.natsConfig.ToConfig().EventTypePrefix,
+			appLister,
+		)
+
+		// Setting up fake informers
+		scheme := runtime.NewScheme()
```

Comment:
Informer set up is duplicated from the testing of `/publish`.

Reply:
Done